pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +137,231 @@ func TestTimeSeriesQuery(t *testing.T) {
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					/* for the following JSON:
+					isMathExpression = true is satisfied by: "metricQueryType": 0, "metricEditorMode": 1, not SEARCH() expression
+					no "alias" */
+					JSON: json.RawMessage(`{
+						"type":      "timeSeriesQuery",
+						"metricQueryType": 0,
+						"metricEditorMode": 1,
+						"namespace": "",
+						"metricName": "",
+						"region": "us-east-2",
+						"id": "query id",
+						"statistic": "Maximum",
+						"period": "1200",
+						"hide": false,
+						"matchExact": true,
+						"refId": "A"
+					}`),
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "query id", resp.Responses["A"].Frames[0].Name)
+	})
+
+	t.Run("where no alias provided and query type is MetricQueryTypeQuery, then frame name is label", func(t *testing.T) {
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					/* for the following JSON:
+					"metricQueryType": 1 is MetricQueryTypeQuery
+					no "alias" */
+					JSON: json.RawMessage(`{
+						"type":      "timeSeriesQuery",
+						"metricQueryType": 1,
+						"metricEditorMode": 0,
+						"namespace": "",
+						"metricName": "",
+						"dimensions": {"InstanceId": ["some-instance"]},
+						"region": "us-east-2",
+						"id": "query id",
+						"statistic": "Maximum",
+						"period": "1200",
+						"hide": false,
+						"matchExact": false,
+						"refId": "A"
+					}`),
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "response label", resp.Responses["A"].Frames[0].Name)
+	})
+}
+
+func Test_QueryData_where_no_alias_provided_then_frame_name_depends_on_dimension_values_and_matchExact(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+	var cwClient fakeCWClient
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return &cwClient
+	}
+	cwClient = fakeCWClient{
+		GetMetricDataOutput: cloudwatch.GetMetricDataOutput{
+			MetricDataResults: []*cloudwatch.MetricDataResult{
+				{StatusCode: aws.String("Complete"), Id: aws.String("query id"), Label: aws.String("response label"),
+					Values: []*float64{aws.Float64(1.0)}, Timestamps: []*time.Time{{}}},
+			},
+		},
+	}
+	im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
+
+	// where query is inferred search expression and not multivalued dimension expression, then frame name is label
+	testCasesReturningLabel := map[string]struct {
```

Comment:
Can this be an array of structs:
```
[]struct {
  name string
  dimensions string
  matchExact bool
} { ...
```
So that the tests run in the same order every time?

Reply:
Ah I don't think tests need to be run in a certain order. Just curious why you think the order of the tests need to be the same every time?

Reply:
Here's a relevant [blurb from Dave Cheney ](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests):
>There’s also a side benefit of using a map that is going to potentially improve the utility of our tests.
Map iteration order is undefined. This means each time we run go test, our tests are going to be potentially run in a different order.
This is super useful for spotting conditions where test pass when run in statement order, but not otherwise. If you find that happens you probably have some global state that is being mutated by one test with subsequent tests depending on that modification.

Reply:
That's also valid. I find that having a regular test order makes it easier for me to scan on re-run but its not super important.